src/custom-element.js

```diff
@@ -1041,9 +1041,14 @@ function createBaseCustomElementClass(win) {
       if (this.perfOn_) {
         this.getLayoutDelayMeter_().startLayout();
       }
-      const promise = this.implementation_.layoutCallback();
-      this.preconnect(/* onLayout */true);
-      this.classList.add('i-amphtml-layout');
+
+      let promise;
+      try {
```

Comment:
you can use: https://github.com/ampproject/amphtml/blob/e21f78c39979f041bd7fda9a2a8c9ea0d937bdad/src/utils/promise.js#L70

Reply:
done